src/main/pages/setup-docker/docker-config.adoc

```diff
@@ -472,56 +497,68 @@ CHE_HTTPS_PROXY_FOR_WORKSPACES=<YOUR_PROXY_FROM_DOCKER>
+The last three entries are injected into workspaces created by your users. This gives your users access to the Internet from within their workspaces. You can comment out these entries to disable access. However, if that access is turned off, the default templates with source code fails to be created in workspaces as those projects are cloned from GitHub. Your workspaces are still functional and only  template cloning is prevented.
 
 On Linux, a firewall may block inbound connections from within Docker containers to your localhost network. As a result, the workspace agent is unable to ping the Che server. You can check for the firewall and then disable it.
 
-Firewalls will typically cause traffic problems to appear when you are starting a new workspace. There are certain network configurations where we direct networking traffic between workspaces and Che through external IP addresses, which can flow through routers or firewalls. If ports or protocols are blocked, then certain functions will be unavailable.
+Firewalls will typically cause traffic problems to appear when you are starting a new workspace. There are certain network configurations where we direct networking traffic between workspaces and Che through external IP addresses that can flow through routers or firewalls. If ports or protocols are blocked, certain functions will be unavailable.
 
-*Running Behind a Firewall (Linux/Mac)*
+*Running behind a firewall (Linux/macOS)*
 
+. To check if firewall is running:
++
 ----
-# Check to see if firewall is running:
 systemctl status firewalld
-
-# Check for list of open ports
-# Verify that ports 8080tcp, 32768-65535tcp are open
+----
++
+. To check the list of open ports, verify that ports `8080tcp`, `32768`-`65535tcp` are open:
++
+----
 firewall-cmd --list-ports
-
-# Optionally open ports on your local firewall:
+----
++
+. Optionally, open ports on your local firewall:
++
+----
 firewall-cmd --permanent --add-port=8080/tcp
-... and so on
-
-# You can also verify that ports are open:
+----
++ 
+Use this command to open other ports, too.
++
+. Optionally, to verify that ports are open:
++
+----
 nmap -Pn -p <port> localhost
+----
++
+If the port is closed, you must open it by editing the `/etc/pf.conf` file. For example, to open port `1234` for TCP for all interfaces, pass in `proto tcp` from any port to any port 1234. Then, restart your firewall.
 
-# If the port is closed, then you need to open it by editing /etc/pf.conf.
-# For example, open port 1234 for TCP for all interfaces:
-pass in proto tcp from any to any port 1234
+*Running Che behind a firewall (Windows)*
 
-# And then restart your firewall
-----
+There are many third-party firewall services. Different versions of Windows OS also have different firewall configurations. The built-in Windows firewall can be configured in the control panel under "System and Security":
```

Comment:
* Windows OS -> Windows
* control panel -> Control Panel
* System and Security -- no quotes but in bold